upload-api/tables/provisions.js

```diff
@@ -0,0 +1,188 @@
+import {
+  DynamoDBClient,
+  QueryCommand,
+  PutItemCommand,
+  DescribeTableCommand,
+} from '@aws-sdk/client-dynamodb'
+import { Failure } from '@ucanto/server'
+import { marshall, } from '@aws-sdk/util-dynamodb'
+
+/**
+ * Abstraction layer to handle operations on Store Table.
+ *
+ * @param {string} region
+ * @param {string} tableName
+ * @param {import('@ucanto/interface').DID<'web'>[]} services
+ * @param {object} [options]
+ * @param {string} [options.endpoint]
+ */
+export function createProvisionsTable (region, tableName, services, options = {}) {
+  const dynamoDb = new DynamoDBClient({
+    region,
+    endpoint: options.endpoint,
+  })
+
+  return useProvisionsTable(dynamoDb, tableName, services)
+}
+
+class ConflictError extends Failure {
+  /**
+   * @param {object} input
+   * @param {string} input.message
+   */
+  constructor({ message }) {
+    super(message)
+    this.name = 'ConflictError'
+  }
+}
+
+/**
+ * @param {DynamoDBClient} dynamoDb
+ * @param {string} tableName
+ * @param {import('@ucanto/interface').DID<'web'>[]} services
+ * @returns {import('@web3-storage/upload-api').ProvisionsStorage}
+ */
+export function useProvisionsTable (dynamoDb, tableName, services) {
+  return {
+    services,
+    hasStorageProvider: async (consumer) => {
+      const cmd = new QueryCommand({
```

Comment:
Looks like we only need one. Can't we use GetItemCommand instead here? I believe it will be more efficient than a query

Reply:
removed this in favor of subscription/consumer tables